src/main/java/ca/on/oicr/gsi/dimsum/CaseLoader.java

```diff
@@ -60,507 +38,108 @@ public class CaseLoader {
-      log.debug("Current case data is up to date with data files; aborting load");
-      return null;
-    }
-    long startTimeMillis = System.currentTimeMillis();
-    try (FileReader projectReader = getProjectReader();
-        FileReader sampleReader = getSampleReader();
-        FileReader donorReader = getDonorReader();
-        FileReader requisitionReader = getRequisitionReader();
-        FileReader runReader = getRunReader();
-        FileReader assayReader = getAssayReader();
-        FileReader caseReader = getCaseReader();
-        FileReader nocaseReader = getNoCaseReader();) {
-      ZonedDateTime afterTimestamp = getDataTimestamp();
-      if (afterTimestamp == null) {
-        log.debug("New case data is currently being written; aborting load.");
-        return null;
-      } else if (!afterTimestamp.equals(beforeTimestamp)) {
-        // Data was written in the middle of loading. Restart
-        log.debug("New case data was written while we were reading; reloading...");
-        return load(previousTimestamp);
-      }
-      Map<String, Project> projectsByName = loadProjects(projectReader);
-      Map<String, Donor> donorsById = loadDonors(donorReader);
-      Map<Long, Run> runsById = loadRuns(runReader);
-      Map<Long, Requisition> requisitionsById = loadRequisitions(requisitionReader, donorsById);
-      Map<String, Sample> samplesById =
-          loadSamples(sampleReader, donorsById, runsById, requisitionsById);
-      List<OmittedSample> omittedSamples =
-          loadOmittedSamples(nocaseReader, donorsById, requisitionsById);
-      Map<Long, Assay> assaysById = loadAssays(assayReader);
-      List<Case> cases = loadCases(caseReader, projectsByName, samplesById, donorsById,
-          requisitionsById, assaysById);
-      Map<String, RunAndLibraries> runsByName = sortRuns(cases);
-      if (refreshTimer != null) {
-        refreshTimer.record(System.currentTimeMillis() - startTimeMillis, TimeUnit.MILLISECONDS);
-      }
-
-      CaseData caseData =
-          new CaseData(cases, runsByName, assaysById, omittedSamples, afterTimestamp,
-              getRequisitionNames(requisitionsById), getProjectNames(projectsByName),
-              getDonorNames(donorsById), getRunNames(runsByName), getTestNames(cases),
-              calculateProjectSummaries(cases, null));
 
-      log.debug(String.format("Completed loading %d cases.", cases.size()));
+    WebClient.Builder builder = WebClient.builder();
+    ZonedDateTime currentTimeStamp = builder.build().get().uri(cardeaUrl + "/timestamp").retrieve()
+        .bodyToMono(ZonedDateTime.class).block();
 
-      return caseData;
+    if (previousTimestamp != null && !currentTimeStamp.isAfter(previousTimestamp)) {
+      log.debug("Current case data is up to date with data files; aborting reload.");
+      return null;
     }
-  }
+    ca.on.oicr.gsi.cardea.data.CaseData cardeaCaseData = loadCardeaData(builder);
```

Comment:
If Cardea returned an empty/null response for whatever reason, should Dimsum go ahead and replace its cache with the empty/null data?

Reply:
Maybe it should just retain the previous data as opposed to the null response, so I could treat it as a case where the logger has an error message describing what happened, and the function returns `null`?

Reply:
I think that's a good plan, and our existing monitoring should alert us for when this happens. You might need to update downstream functions that consume cardeaCaseData, so they don't run into NPEs.